code/modules/clothing/masks/boxing.dm

```diff
@@ -11,6 +11,38 @@
 /obj/item/clothing/mask/balaclava/attack_self(mob/user)
 	adjustmask(user)
 
+/obj/item/clothing/mask/infiltrator
+	name = "infiltrator balaclava"
+	desc = "It makes you feel safe in your anonymity, but for a stealth outfit you sure do look obvious that you're up to no good. It seems to have a built in heads-up display."
+	icon_state = "syndicate_balaclava"
+	item_state = "syndicate_balaclava"
+	clothing_flags = MASKINTERNALS
+	flags_inv = HIDEFACE|HIDEHAIR|HIDEFACIALHAIR
+	visor_flags_inv = HIDEFACE|HIDEFACIALHAIR
+	w_class = WEIGHT_CLASS_SMALL
+	armor = list("melee" = 10, "bullet" = 5, "laser" = 5,"energy" = 5, "bomb" = 0, "bio" = 0, "rad" = 10, "fire" = 100, "acid" = 40)
+	resistance_flags = FIRE_PROOF | ACID_PROOF
+	
+	var/vunknown = FALSE ///This makes it so that your name shows up as unknown when wearing the mask.
+
+/obj/item/clothing/mask/infiltrator/equipped(mob/living/carbon/human/user, slot)
+	..()
+	if(slot != ITEM_SLOT_MASK)
+		return
+	to_chat(user, "You roll the balaclava over your face, and a data display appears before your eyes.")
+	ADD_TRAIT(user, TRAIT_DIAGNOSTIC_HUD, MASK_TRAIT)
+	var/datum/atom_hud/H = GLOB.huds[DATA_HUD_DIAGNOSTIC_ADVANCED]

```

Comment:
You may wish to use the BASIC version here. ADVANCED is the silicon version, and the key difference is that ADVANCED shows bot routes (such as Beepsky's path to the next patrol beacon). Human players would probably consider it clutter, so I do not recommend using it.

Reply:
noted, cheers